### PR TITLE
Add more type hinting to the TestCase constructor

### PR DESCRIPTION
**Pull request by oliverklee**

### CODE REVIEW

**Comment by oliverklee** [comment]
I don't know whether tests/_files/ModifiedConstructorTestCase.php needs to be adapted, though.

File: src/Framework/TestCase.php [hunk 1]
```diff
@@ -375,10 +375,9 @@ public static function onConsecutiveCalls(...$args): ConsecutiveCallsStub
     }
 
     /**
-     * @param string $name
      * @param string $dataName
      */
-    public function __construct($name = null, array $data = [], $dataName = '')
+    public function __construct(?string $name = null, array $data = [], $dataName = '')
```
Comment (oliverklee) [inline src/Framework/TestCase.php]: This is only one small change towards improving the declaration. More PRs for this method will come.

**Comment by sebastianbergmann** [comment]
This has to wait until PHPUnit 9 as it breaks backward compatibility (not really, but technically).

**Comment by oliverklee** [comment]
@sebastianbergmann When will master become "9-to-be", e.g., when will the last 8.x release from master be released?

**Comment by sebastianbergmann** [comment]
After PHPUnit 8.5 has been released in December. Don't worry, I'll take care of it then.

**Comment by oliverklee** [comment]
Thanks!

**Pull request merged**